src/Microsoft.ApplicationInsights.AspNet/RequestTrackingMiddleware.cs

```
@@ -32,31 +29,33 @@ public RequestTrackingMiddleware(RequestDelegate next, TelemetryClient client)
 
         public async Task Invoke(HttpContext httpContext, RequestTelemetry telemetry)
         {
+            telemetry.Timestamp = DateTimeOffset.UtcNow;
+
             var sw = new Stopwatch();
             sw.Start();
 
-            var now = DateTimeOffset.UtcNow;
+            bool requestFailed = false;
 
             try
             {
                 await this.next.Invoke(httpContext);
             }
+            catch (Exception)
+            {
+                requestFailed = true;
```

should we rethrow this exception?